src/mlir/dialect/VeronaOps.cc

```diff
@@ -111,6 +133,32 @@ static LogicalResult verify(verona::ClassOp classOp)
   return success();
 }
 
+static LogicalResult verify(verona::WhileOp whileOp)
+{
+  auto& body = whileOp.body();
```

Comment:
I think you need to verify that the region has no parameters (eg. using NoRegionArguments).
It's not allowed by the parser, by the generic syntax would let you construct one (or through builders).

I guess with value tracking this requirement will change, but we should still encode it now.